### DISCUSSION TITLE
Heap/Stack/RAM question

### DISCUSSION THREAD

**Discussion by git2212** [earlephilhower/arduino-pico#1435] [General]
Are the values in `__brkval` and `_estack` reliable/accurate at runtime on this core? 

Thx

**Comment by earlephilhower** [comment]
What are you trying to do with them?  If you need the total free/used heap, check https://arduino-pico.readthedocs.io/en/latest/rp2040.html#memory-information

**Comment by git2212** [comment]
Already using the helper class ;)... under legacy Arduino, these two variables reliably give heap and stack info... was wondering if this was carried forward for the RP2040?

**Reply by earlephilhower** [reply]
We use Newlib's malloc implementation coupled with the Pico-SDK runtime which do not use those variables, sorry.

**Comment by git2212** [comment]
ok thanks.
